frontend/packages/client/src/components/Settings/NotificationSettingsSection/EmailAddressInput.js

```diff
@@ -20,8 +20,12 @@ export default function EmailAddressInput({ email, setUserEmail }) {
       })
     ),
   });
-  const onSubmit = ({ email }) => {
-    setUserEmail(email);
+  const onSubmit = async ({ email }) => {
+    try {
+      await setUserEmail(email);
+    } catch (e) {
+      setValue('email', defaultEmail);
```

Comment:
the `setUserEmail` is wrapped by the `hadleError` high order function defined in `NotificationSettingsSection/index.js` line 18. If there's an error occurred in the api call, it will first be caught by the handleError function, error modal will show up with the message, then it will be re-thrown and caught by the catch block here, which reverts the value of the email address input since the setting wasn't successful. It's for sure a lot, hope it makes sense 😆

Reply:
oh i see, missed that. Thanks for the explanation! 👍